site/src/components/form.md

```diff
@@ -0,0 +1,176 @@
+		/>
+	</div>
+</div>
+<div class="field">
+	<div class="field-section">
+		<label class="field-section-label field-section-label-lg label" for="sample-form-input-lg">
+			Input
+		</label>
+		<input
+			id="sample-form-input-lg"
+			name="title"
+			class="input"
+			type="text"
+			placeholder="e.g., Why is the sky blue?"
+		/>
+	</div>
+</div>
+<div class="field display-flex">
+	<div class="field-section">
+		<button class="button button-primary margin-right-xxs">Submit</button>
+		<button class="button button-danger">Cancel</button>
+	</div>
+</div>
+```
+
+### Modifiers
+
+`.field-addons` is added to attach controls together.
+
+```html
+<div class="field field-addons">
+	<div class="field-section">
+		<div class="select">
+			<select>
+				<option>$</option>
+				<option>€</option>
+			</select>
+		</div>
+	</div>
+	<div class="field-section">
+		<input class="input" type="text" placeholder="Transfer money" />
+	</div>
+	<div class="field-section">
+		<a class="button button-primary"> Transfer </a>
+	</div>
+</div>
+```
+
+## Form Section
+
+Form control elements are wrapped with a `.field-section` container.
+
+`.field-section-message` is placed after the element for form validation messages.
+
+```html
+<div class="field">
+	<div class="field-section">
+		<label class="field-section-label field-section-label-lg label" for="sample-form-input-note">
+			Input
+		</label>
+		<input
+			id="sample-form-input-note"
+			name="title"
+			class="input"
+			type="text"
+			placeholder="e.g., Why is the sky blue?"
+			aria-describedby="section-note"
+		/>
+		<p id="section-note" class="field-section-message help help-danger">Sample warning message.</p>
+	</div>
+</div>
+```
+
+### Modifiers
+
+`.field-section-icons` is used when the control element contains icons.
+
+```html
+<div class="field">
+	<div class="field-section field-section-icons">
+		<input class="input input-sm" placeholder="www.microsoft.com" />
+		<span class="icon border" aria-hidden="true"> </span>
+	</div>
+	<div class="field-section field-section-icons margin-top-xxs">
+		<div class="select select-sm">
+			<select>
+				<option>A</option>
+				<option>B</option>
+			</select>
+			<span class="icon border" aria-hidden="true"> </span>
+		</div>
+	</div>
+</div>
+```
+
+`.field-section-icon-right` is used to display the icon on the right side of the element.
+
+```html
+<div class="field">
+	<div class="field-section field-section-icons field-section-icon-right">
+		<input class="input input-sm" placeholder="www.microsoft.com" />
+		<span class="icon border" aria-hidden="true"> </span>
+	</div>
+</div>
+```
+
+### Required Indicator
+
+For control elements that are required for form submission, you can add a span element with the `.required-indicator` class inside the label or adjacent to the element.
+
+```html
+<div class="field">
+	<p class="margin-bottom-xs">
+		Required fields are marked with an
+		<span class="required-indicator" aria-label="required"></span>
+	</p>
```

Comment:
I know you're reviewing this with the accessibility team, just wanted to mention that this reads as "Required fields are marked with an required".

Reply:
Ooh, this one is an interesting case.

Jeremy, is there a particular browser/screenreader combo you're seeing this announcement in? `aria-label` is only intended for interactive elements and landmarks, so while some browser/screenreader pairs might announce the label, that's definitely nonstandard, and I believe _most_ browser/screenreader combos would announce the span's contents.

The accessibility team recommended not doing anything special for the asterisks' markup -- just `<span class="required-indicator">` -- since the fields' `required` attributes will convey the required status. But in this case, where we have some help text disconnected from any inputs, I do think we might actually want to consider doing some markup trickery. I say this because [while JAWS and VoiceOver both announce "star" for asterisks by default, NVDA omits them entirely by default](https://www.deque.com/blog/dont-screen-readers-read-whats-screen-part-1-punctuation-typographic-symbols/#grammatical). Things get even trickier when folks have tweaked their verbosity settings. Because of this, when NVDA users and anyone with low-verbosity screenreader settings navigate to this help text, they'll likely be told `Required fields are marked with an`.

I might recommend something like:

```html
<p class="margin-bottom-xs">
	Required fields are marked with an
	<span class="required-indicator" aria-hidden="true"></span>
	<span class="visually-hidden">asterisk</span>
</p>
```

in this particular case.

Reply:
I think it was reading that way because of the aria-label. For me, NVDA does read "star" as expected after removing the aria-label, but I've never changed my verbosity settings so I suppose I should explore that.

Reply:
Removed